### DISCUSSION TITLE
Using STM32F407IGT6 in the ARDUINO framework

### DISCUSSION THREAD

**Discussion by camerakang** [stm32duino/Arduino_Core_STM32#2583] [Q&A] [answered]
I have added the file to the following path：

> C:\Users\OSD\.platformio\packages\framework-arduinoststm32\variants\STM32F4xx\F407I(E-G)(H-T)_F417I(E-G)(H-T)
![image](https://github.com/user-attachments/assets/9a80d025-dfd5-4463-9430-84a5ba1987ba)

> C:\Users\OSD\.platformio\platforms\ststm32\boards
![Uploading image.png…]()

> C:\Users\OSD\.platformio\packages\framework-arduinoststm32\boards.txt
`# VCCGND_F407IGT6 board
GenF4.menu.pnum.VCCGND_F407IG_MINI=VCCGND F407IGT6 Mini
GenF4.menu.pnum.VCCGND_F407IG_MINI.upload.maximum_size=1048576
GenF4.menu.pnum.VCCGND_F407IG_MINI.upload.maximum_data_size=131072
GenF4.menu.pnum.VCCGND_F407IG_MINI.build.board=VCCGND_F407IG_MINI
GenF4.menu.pnum.VCCGND_F407IG_MINI.build.product_line=STM32F407xx
GenF4.menu.pnum.VCCGND_F407IG_MINI.build.variant_h=variant_{build.board}.h
GenF4.menu.pnum.VCCGND_F407IG_MINI.build.variant=STM32F4xx/F407I(E-G)(H-T)_F417I(E-G)(H-T)
GenF4.menu.pnum.VCCGND_F407IG_MINI.build.peripheral_pins=-DCUSTOM_PERIPHERAL_PINS`

However, the following errors occurred during compilation after creating the new project:
`C:\Users\OSD\.platformio\penv\Scripts\platformio.exe run 

Processing vccgnd_f407Ig_mini (platform: ststm32; board: vccgnd_f407Ig_mini; framework: arduino)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/ststm32/vccgnd_f407Ig_mini.html
PLATFORM: ST STM32 (17.2.0) > VCCGND F407IGT6 Mini
HARDWARE: STM32F407ZGT6 168MHz, 128KB RAM, 1MB Flash
DEBUG: Current (blackmagic) External (blackmagic, cmsis-dap, jlink, stlink)
PACKAGES:
 - framework-arduinoststm32 @ 4.20701.0 (2.7.1)
 - framework-cmsis @ 2.50900.0 (5.9.0)
 - toolchain-gccarmnoneeabi @ 1.120301.0 (12.3.1)
Warning! Cannot find linker script for the current target!

LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 12 compatible libraries
Scanning dependencies...
No dependencies
Building in release mode
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_adc.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_adc_ex.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_can.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cec.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_comp.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_comp_ex.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cordic.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cortex.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_crc.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_crc_ex.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cryp.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cryp_ex.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dac.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dac_ex.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dcache.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dcmi.c.o
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dcmi_ex.c.o
https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dfsdm.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dfsdm_ex.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dma.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dma2d.c.o
Compiling .pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dma_ex.c.o
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_adc.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_adc_ex.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cec.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_can.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cortex.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cryp.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_crc.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_cryp_ex.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dma.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dma2d.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dcmi.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dma_ex.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dac.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dcmi_ex.c.o] Error 1
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dfsdm.c.o] Error 1
<command-line>: fatal error: variant_VCCGND_F407IG_MINI.h: No such file or directory

************************************************************************************
* Looking for variant_VCCGND_F407IG_MINI.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:variant_VCCGND_F407IG_MINI.h"
* Web  > https://registry.platformio.org/search?q=header:variant_VCCGND_F407IG_MINI.h
*
************************************************************************************

compilation terminated.
*** [.pio\build\vccgnd_f407Ig_mini\SrcWrapper\src\HAL\stm32yyxx_hal_dac_ex.c.o] Error 1`

May I ask what I did wrong?

**Comment by camerakang** [comment]
![image](https://github.com/user-attachments/assets/aba00755-6f87-411e-a424-7987db5b87af)
![image](https://github.com/user-attachments/assets/5aa40019-fe56-499a-9a06-661977428cfe)
![image](https://github.com/user-attachments/assets/44af6400-3cf6-44ac-ab52-e020d32e5681)
This is the file I added

**Comment by fpistm** [comment]
Hi @camerakang 

> May I ask what I did wrong?

I don't know as we don't support PIO here nor know how PIO integrate this core.
You should file an issue on PIO.
Anyway, I guess they will tell you to firstly open a PR here to add the support of this board then when merged and released, PIO will update with the new release.

**Reply by camerakang** [reply, marked as answer]
Thank you very much for your reply, I will go to the PIO forum and reask the question!
